### ISSUE TITLE
Inconsistent Results between TensorRT and ONNX Inference with Split Operator

### ISSUE THREAD

**Issue by hongliyu0716** [NVIDIA/TensorRT#3466]
## Description

When performing inference with an ONNX model that includes the Split operator using TensorRT, the inference results obtained using TensorRT differ from the results obtained using ONNX runtime.
The ONNX model structure is as below:
![Split onnx](https://github.com/NVIDIA/TensorRT/assets/149884162/3b4b98fb-445e-4d00-9b9f-bdaef1f2f249)

## Environment

**TensorRT Version**: 8.6.1.6

**NVIDIA GPU**: NVIDIA GeForce MX330

**NVIDIA Driver Version**: 470.182.03

**CUDA Version**: 11.4

**CUDNN Version**: 8.9.5

Operating System: Ubuntu 18.04

Python Version (if applicable): 3.8

## Relevant Files

**Model link**: https://github.com/hongliyu0716/onnx_model/blob/main/Split.onnx

## Steps To Reproduce

1. Download the model
2. **Commands or scripts**:
```
polygraphy run Split.onnx --onnxrt --trt --workspace 256M --save-engine test.plan --fp16 --verbose
``` 
The error message is as below:
```
[I] trt-runner-N0-11/18/23-19:41:20    
    ---- Inference Input(s) ----
    {input [dtype=float32, shape=(7,)]}
[V] trt-runner-N0-11/18/23-19:41:20     | Input metadata is: {input [dtype=float32, shape=(7,)]}
[I] trt-runner-N0-11/18/23-19:41:20    
    ---- Inference Output(s) ----
    {output_1 [dtype=float32, shape=(1,)],
     output_2 [dtype=float32, shape=(1,)],
     output_3 [dtype=float32, shape=(1,)],
     output_4 [dtype=float32, shape=(1,)]}
[I] trt-runner-N0-11/18/23-19:41:20     | Completed 1 iteration(s) in 0.8709 ms | Average inference time: 0.8709 ms.
[V] Successfully ran: ['onnxrt-runner-N0-11/18/23-19:41:20', 'trt-runner-N0-11/18/23-19:41:20']
[I] Accuracy Comparison | onnxrt-runner-N0-11/18/23-19:41:20 vs. trt-runner-N0-11/18/23-19:41:20
[I]     Comparing Output: 'output_1' (dtype=float32, shape=(2,)) with 'output_1' (dtype=float32, shape=(1,))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[E]         Will not compare outputs of different shapes. Note: Output shapes are (2,) and (1,).
[E]         Note: Use --no-shape-check or set check_shapes=False to attempt to compare values anyway.
[E]         FAILED | Output: 'output_1' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'output_2' (dtype=float32, shape=(2,)) with 'output_2' (dtype=float32, shape=(1,))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[E]         Will not compare outputs of different shapes. Note: Output shapes are (2,) and (1,).
[E]         FAILED | Output: 'output_2' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'output_3' (dtype=float32, shape=(2,)) with 'output_3' (dtype=float32, shape=(1,))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[E]         Will not compare outputs of different shapes. Note: Output shapes are (2,) and (1,).
[E]         FAILED | Output: 'output_3' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[I]     Comparing Output: 'output_4' (dtype=float32, shape=(1,)) with 'output_4' (dtype=float32, shape=(1,))
[I]         Tolerance: [abs=1e-05, rel=1e-05] | Checking elemwise error
[I]         onnxrt-runner-N0-11/18/23-19:41:20: output_4 | Stats: mean=0.18626, std-dev=0, var=0, median=0.18626, min=0.18626 at (0,), max=0.18626 at (0,), avg-magnitude=0.18626
[I]             ---- Values ----
                    [0.18626021]
[I]             ---- Histogram ----
                Bin Range      |  Num Elems | Visualization
                (0.186, 0.198) |          1 | ########################################
                (0.198, 0.209) |          0 | 
                (0.209, 0.221) |          0 | 
                (0.221, 0.233) |          0 | 
                (0.233, 0.244) |          0 | 
                (0.244, 0.256) |          0 | 
                (0.256, 0.268) |          0 | 
                (0.268, 0.279) |          0 | 
                (0.279, 0.291) |          0 | 
                (0.291, 0.302) |          0 | 
[I]         trt-runner-N0-11/18/23-19:41:20: output_4 | Stats: mean=0.30233, std-dev=0, var=0, median=0.30233, min=0.30233 at (0,), max=0.30233 at (0,), avg-magnitude=0.30233
[I]             ---- Values ----
                    [0.30233258]
[I]             ---- Histogram ----
                Bin Range      |  Num Elems | Visualization
                (0.186, 0.198) |          0 | 
                (0.198, 0.209) |          0 | 
                (0.209, 0.221) |          0 | 
                (0.221, 0.233) |          0 | 
                (0.233, 0.244) |          0 | 
                (0.244, 0.256) |          0 | 
                (0.256, 0.268) |          0 | 
                (0.268, 0.279) |          0 | 
                (0.279, 0.291) |          0 | 
                (0.291, 0.302) |          1 | ########################################
[I]         Error Metrics: output_4
[I]             Minimum Required Tolerance: elemwise error | [abs=0.11607] OR [rel=0.38392] (requirements may be lower if both abs/rel tolerances are set)
[I]             Absolute Difference | Stats: mean=0.11607, std-dev=0, var=0, median=0.11607, min=0.11607 at (0,), max=0.11607 at (0,), avg-magnitude=0.11607
[I]                 ---- Values ----
                        [0.11607237]
[I]                 ---- Histogram ----
                    Bin Range          |  Num Elems | Visualization
                    (-0.384 , -0.284 ) |          0 | 
                    (-0.284 , -0.184 ) |          0 | 
                    (-0.184 , -0.0839) |          0 | 
                    (-0.0839, 0.0161 ) |          0 | 
                    (0.0161 , 0.116  ) |          0 | 
                    (0.116  , 0.216  ) |          1 | ########################################
                    (0.216  , 0.316  ) |          0 | 
                    (0.316  , 0.416  ) |          0 | 
                    (0.416  , 0.516  ) |          0 | 
                    (0.516  , 0.616  ) |          0 | 
[I]             Relative Difference | Stats: mean=0.38392, std-dev=0, var=0, median=0.38392, min=0.38392 at (0,), max=0.38392 at (0,), avg-magnitude=0.38392
[I]                 ---- Values ----
                        [0.38392282]
[I]                 ---- Histogram ----
                    Bin Range          |  Num Elems | Visualization
                    (-0.116 , -0.0161) |          0 | 
                    (-0.0161, 0.0839 ) |          0 | 
                    (0.0839 , 0.184  ) |          0 | 
                    (0.184  , 0.284  ) |          0 | 
                    (0.284  , 0.384  ) |          0 | 
                    (0.384  , 0.484  ) |          1 | ########################################
                    (0.484  , 0.584  ) |          0 | 
                    (0.584  , 0.684  ) |          0 | 
                    (0.684  , 0.784  ) |          0 | 
                    (0.784  , 0.884  ) |          0 | 
[E]         FAILED | Output: 'output_4' | Difference exceeds tolerance (rel=1e-05, abs=1e-05)
[E]     FAILED | Mismatched outputs: ['output_1', 'output_2', 'output_3', 'output_4']
[E] Accuracy Summary | onnxrt-runner-N0-11/18/23-19:41:20 vs. trt-runner-N0-11/18/23-19:41:20 | Passed: 0/1 iterations | Pass Rate: 0.0%
[E] FAILED | Runtime: 5.796s | Command: /home/hll/anaconda3/envs/trt/bin/polygraphy run Split.onnx --onnxrt --trt --workspace 256M --save-engine test.plan --fp16 --verbose
```

**Comment by zhenhuaw-me** [comment]
The ONNX spec has a bug. See https://github.com/onnx/onnx/issues/5766

**Comment by zerollzeng** [comment]
Issue will be fixed in TRT 10.0, close this.
